.github/actions/configure-wkg/action.yml

```diff
@@ -0,0 +1,30 @@
+name: Install and configure wkg (linux only)
+
+inputs:
+  wkg-version:
+    description: version of wkg to install. Should be a valid tag from https://github.com/bytecodealliance/wasm-pkg-tools/releases
+    default: "v0.7.0"
+
+runs:
+  using: composite
+  steps:
+    - name: Download wkg
+      shell: bash
+      run: |
+        curl -L https://github.com/bytecodealliance/wasm-pkg-tools/releases/download/${{ inputs.wkg-version }}/wkg-x86_64-unknown-linux-gnu -o wkg
+        chmod +x wkg;
+        echo "$(realpath .)" >> "$GITHUB_PATH";
+    - name: Generate and set wkg config
+      shell: bash
+      env:
+        WKG_OCI_USERNAME: ${{ github.repository_owner }}
+        WKG_OCI_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        cat << EOF > wkg-config.toml
+        [namespace_registries]
+        wasmcloud = "wasmcloud.com"
+
+        [registry."wasmcloud.com".oci]
+        auth = { username = "${WKG_OCI_USERNAME}", password = "${WKG_OCI_PASSWORD}" }
+        EOF
+        echo "WKG_CONFIG_FILE=$(realpath wkg-config.toml)" >> $GITHUB_ENV
```

Comment:
I’m not sure I follow. This is a temporary config just when configuring wkg so I’m not sure it matters where we put it. Putting it in a specific package doesn’t really change that. My reasoning here is configure wkg to just work when someone uses it from there on out

Reply:
I'm not sure if it makes sense to keep this config separate from the code that it's related to. For example, if we were building a component/provider, we wouldn't generate a temporary `wasmcloud.toml` right? We'd put the `wasmcloud.toml` next to the project files because it's related/configures the use of tooling with the project.

I think of the `wasmcloud/wit` folders as the same (ex. `wasmcloud/wit/postgres`) -- there's a README.md in there and the wit is subfoldered so that we can sort of create a consistent entrypoint for the WIT package.
`wkg-config.toml` seems to be tooling that is *related* to that package, so I was thinking that it should be there, with the actual package.

As far as the CI goes, avoiding implicit behavior like writing files and setting new ENV would probably make the resulting workflows that depend on the action a bit clearer. It *does* make the dependent actions a little more wordy, but there's a lot less action-at-a-distance, I think!

Reply:
This config isn't related specifically to this code. Put another way, if this config were on a user's machine, it would be in their XDG config directory and all projects would use it to push. All this action is supposed to be doing is setting up `wkg` to be able to publish anything it needs to. So it is less akin to a wasmcloud.toml and more akin to a kubeconfig

> As far as the CI goes, avoiding implicit behavior like writing files and setting new ENV would probably make the resulting workflows that depend on the action a bit clearer. It does make the dependent actions a little more wordy, but there's a lot less action-at-a-distance, I think!

This would be a lot of copy/pasting and knowing what vars to configure IMO.

Reply:
Ahhh so it's a *global* config, that makes a lot of sense then, thanks!

> This would be a lot of copy/pasting and knowing what vars to configure IMO.

Yeah, the tradeoff here makes sense if this file is *normally* one a user might have configured globally.